### DISCUSSION TITLE
Debug run on CPU

### DISCUSSION THREAD

**Discussion by VincentCoulombe** [open-mmlab/mmengine#1510] [Q&A]
Hi,

I have a bug in my data processing pipeline, and it is very hard to debug as all the tensors are put on the GPU automatically.

Reading through the documentation, I have not found how to force the run to be on CPU so that I can have a meaningful traceback when an error is raised.

Does anybody know how?

Thank you,
Vincent

**Comment by AwePhD** [comment]
Hi,

[Here](https://mmdetection.readthedocs.io/en/latest/user_guides/tracking_train_test.html#train-on-cpu) in the doc of mmdet.
Namely, use CUDA environment variable to hide your GPU to torch/mmengine interface with `CUDA_VISIBLE_DEVICES=-1`.
